export/mlab_export_test.py

```diff
@@ -0,0 +1,492 @@
+    expected_value = {'sample': [
+                       {'timestamp': 1, 'value': 10.0},
+                       {'timestamp': 2, 'value': 11.0},
+                       {'timestamp': 3, 'value': 12.0}],
+                      'metric': 'network.ipv4.bytes.rx',
+                      'hostname': 'mlab2.nuq0t',
+                      'experiment': 'utility.mlab'}
+
+    returned_value = mlab_export.get_json_record(
+        'mlab2.nuq0t', 'utility.mlab', 'network.ipv4.bytes.rx', timestamps,
+        values)
+
+    self.assertEqual(returned_value, expected_value)
+    self.assertTrue(mock_logging.called)
+
+  def testunit_write_json_record(self):
+    output = StringIO.StringIO()
+    record = {'sample': [{'timestamp': 1, 'value': 10.0}],
+              'hostname': 'mlab2.nuq0t',
+              'experiment': 'utility.mlab',
+              'metric': 'network.ipv4.bytes.rx'}
+    pretty_json = None
+    expected_value = ('{"sample": [{"timestamp": 1, "value": 10.0}], '
+                      '"metric": "network.ipv4.bytes.rx", "hostname":'
+                      ' "mlab2.nuq0t", "experiment": "utility.mlab"}\n')
+
+    mlab_export.write_json_record(output, record, pretty_json)
+    returned_value = output.getvalue()
+ 
+    self.assertEqual(returned_value, expected_value)
+
+  def testunit_get_rrd_files(self):
+    rrd_dir = os.path.join(os.path.abspath(self._testdata_dir), 'rrd')
+    expected_files = [rrd_dir + '/mlab2.nuq0t/file2.rrd',
+                      rrd_dir + '/mlab2.nuq0t/file1.rrd']
+
+    returned_files = [ fname for fname in mlab_export.get_rrd_files(rrd_dir) ]
+
+    self.assertEqual(returned_files, expected_files)
+
+  @mock.patch('mlab_export.get_rrd_files')
+  @mock.patch('mlab_export.rrdtool.fetch')
+  @mock.patch('mlab_export.get_canonical_names')
+  def testunit_rrd_list(
+      self, mock_get_canonical_names, mock_rrdtool_fetch, mock_get_rrd_files):
+    mock_get_rrd_files.return_value = ['file1']
+    mock_rrdtool_fetch.return_value = (None, ['value'], None)
+    mock_options = mock.Mock()
+    mock_options.rrddir_prefix = os.path.join(self._testdata_dir, 'rrd')
+    mock_options.ts_start = 123
+    mock_options.ts_end = 456
+
+    mlab_export.rrd_list(mock_options)
+
+    mock_get_canonical_names.assert_called_with('file1', 'value', mock_options)
+
+  @mock.patch('mlab_export.get_rrd_files')
+  @mock.patch('mlab_export.rrdtool.fetch')
+  @mock.patch('mlab_export.get_canonical_names')
+  @mock.patch('gzip.open')
+  @mock.patch('StringIO.StringIO.close')
+  def testunit_rrd_export(
+      self, mock_close, mock_open, mock_get_canonical_names, mock_rrdtool_fetch,
+      mock_get_rrd_files):
+    str_fd = StringIO.StringIO()
+    mock_open.return_value = str_fd
+    mock_get_rrd_files.return_value = ['file1']
+    mock_rrdtool_fetch.return_value = [
+        (0, 10, 10), ['value', 'skipped_metric'], [(0.0,0.0)]]
+    mock_get_canonical_names.side_effect = [
+        ('host', 'exp', 'metric'), ('host', 'exp', None)]
```

Comment:
That's a good suggestion. The names are different from the keys: 7879a698521c4aba1066892690d2e96b4064fb60